tools/custom-action/action.yml

```diff
@@ -0,0 +1,5 @@
+name: 'gpuweb-custom-action'
+description: 'This action runs a container with all necessary tools to validate source'
+runs:
+  using: 'docker'
+  image: 'docker://ghcr.io/gpuweb/gpuweb:main'
```

Comment:
Does this mean this actual always uses the image from `main` and not from the PR being tested? It may be annoying to maintain the image if it has to get merged every time we want to see if a fix unblocked PRs. I don't know if this is fixable though, if it's not easy to fix then I guess it's fine.

Reply:
@kainino0x That's true. When creating the flow, I thought that this image should be really, really stable for quite a long time, except for taking in some new diagramming tools if necessary, and maybe only change is moving to 24.04 LTS when that releases. The fixes should mainly go to Makefiles than either entrypoint or Dockerfile, since these are just scaffolds to give a better CI or dev environment than being the actual deal. And any change to Makefiles or scripts is tested when this container image is fired, it does not override those at all.

If every PR built their own image, then the caching compel would go away. One way we can make sure a change to entrypoint.sh or Dockerfile does fix without going through whole loop is by running a script during container build time which copies Git repository and runs the whole entrypoint on it.

Reply:
We have to upgrade bikeshed pretty regularly. W3C's spec-prod enforces that our build always works on the latest bikeshed version. And we always have to work with the latest specref data for cross-linking.

I guess we could move the bikeshed installation and `bikeshed update` out of the image so the image will be much more stable?

Reply:
@kainino0x Is the every-Friday frequency of cron not good enough? If so, we can move it to entrypoint.sh to increase stability but I'd appreciate your guidance on whether cron is good or not for that purpose.

(BTW one trick which boosts image build times is making layers, such as *-zero (only OS packages), *-base (only stable installs), *-x (with non-versionables) but this might be a secondary step)

Reply:
Every Friday isn't enough when there's build breakage and we need to get up to speed with the latest bikeshed and specref, but since we can trigger manual rebuilds it's most likely fine. Somewhat frequent rebuilds are fine, I just wanted to point out it's not really going to be that "stable" in practice.

Since spec-prod already forces us on the latest bikeshed anyway, and specref can't be pinned, I do wonder if we should just have the image build use the latest bikeshed instead of a pinned version?

I don't think we need to optimize the image build time. It seems like it should be pretty fast anyway. TBH I'm not even that concerned about optimizing our PR check times. They're kind of slow, but the only serious issue is the flakiness.

Reply:
@kainino0x I have moved entire Bikeshed logic out of Dockerfile creation and rather made it part of prepare script. Bikeshed is still versioned due to issues in upstream but update runs every time for sure independent of container build timepoint.

Reply:
Removed versioning now thanks to the fix!